common-arcana.lic

```diff
@@ -137,9 +137,33 @@ module DRCA
     activated
   end
 
-  def prepare?(abbrev, mana, symbiosis = false, command = 'prepare', tattoo_tm = false, runestone_name = nil, runestone_tm = false)
+  def prepare?(settings, abbrev, mana, symbiosis = false, command = 'prepare', tattoo_tm = false, runestone_name = nil, runestone_tm = false, unleash_item = false)
     return false unless abbrev
 
+    # minimum settings to handle unleash items
+    # 1. yaml entry of unleash spells, their items, and containers: unleash: list of spells with parameters: item, container
+    # 2. safety function for unleashing the right scroll by "glance #{unleash.spell.unleash_item} in #{unleash.unleash_container}"
+    # 3. option to integrate spellbook with unleash at that point with like "unleash.spell.is_spellbook = true"
+    # 4. special spellbook logic from this
+    if unleash_item
+      # TODO: This makes me think that any ritual spell should have a manifest with: container, item...
+      # TODO: maybe there should be a helper to get a scroll with a particular read, as well, or at least safety check the read
+      if not DRCI.in_hands?(unleash_item)
+        # TODO: this is not normal, if we can get the logic inside this module, then these items can be used in buff scripts
+        # TODO: however, one excellent case for this is the spellbook, which requires travel, special logic, and more yaml settings
+        # TODO: Spellbook management is not a good fit for this module, but maybe it should be called in this clause
+        echo("*** WARNING: Unleash item management must be handled outside of the cast_spell function.")
+        return false
+      end
+      cast_action = "cast my #{unleash_item}" # nb: ruby doesn't interpolate inside the data payload
+      # TODO: Unleash can be optimized based on the cost of the spell, requires an overhaul to discern logic based on skill+spell
+      # TODO: add a prep action for tattoo from settings
+      unleash_data = {"abbrev" => "unleash", "mana" => 15, "cast" => cast_action}
+      cast_spell(unleash_data, settings)
```

Comment:
I'm not a fan of a code path that has `prepare?` call `cast_spell`

Reply:
Agree. I’m also not really following what this change is trying to do and why

Reply:
From reading https://elanthipedia.play.net/Unleash, it seems like you need to cast two spells: unleash and then the scroll spell. 

I recommend adding a new method that understands that’s what needs to happen and it’d make two calls to cast_spell — one for unleash and one for the scroll spell

Disclaimer, I don’t have unleash. This is my suggestion for how to reuse existing code rather than be casting spells in the prepare method.

Reply:
Can you elaborate? This is a very appropriate usage for cast spell.  The function contexts are separate so there is no scope concern.

Reply:
Just my two cents - I almost feel like cast_spell (or somewhere up the chain) needs to be refactored to decide what kind of thing we're doing, and split out from there.

It currently looks like this one method would handle unleashing a tattoo or runestone, neither of which makes any sense, and it's just getting more and more bloat.

As to the fact that you can 'unleash and unleash scroll to unleash a <other> scroll', this feels like classic "you got so caught up in figuring out if you could, you never stopped to consider if you should" requirements.

Reply:
I may take on a cast spell refactor this week.